### DISCUSSION TITLE
insert date in database

### DISCUSSION THREAD

**Discussion by 5011-5011** [nextgenhealthcare/connect#5527] [Q&A] [answered]
I have this little script to format the date I receive, however when I run the channel it saves something else.
I don't know if the problem is in the key value assignment of the mapping variable, or if it is the script.
If anyone has done it somehow, I'd appreciate some guidance.
Thank you very much

![image](https://user-images.githubusercontent.com/98954172/201662276-29e72e58-1f3c-4319-bde5-3dcf4934f8bd.png)

**the data that comes to me is a date but without format for example like this: "19881214" and I need this format: "1988-12-14"**

**Comment by siddharthsharma9258** [comment, marked as answer]
If that is your script, then it is wrong. 
You are putting the word 'date' in the channel variable fnac.
`msg['PID']['PID.7']['PID.7.1']` will not give you the value of the date you are expecting, but its XML representation.

**Reply by pacmano1** [reply]
What exactly are you trying to do @5011-5011 ?

Just in plain terms.

As @siddharthsharma9258 metioned you are not grabbing the string value of the hl7 component since your are missing `.toString()`

Are you trying to create a new date object for calculations?   Trying to just format the date?

**Reply by 5011-5011** [reply]
I am trying to convert to Date. go from "19885425" to "1988-54-25" in order to insert that value into the database. However I don't understand how the mapping variable works

**Reply by pacmano1** [reply]
Firstly:

![image](https://user-images.githubusercontent.com/44065187/201733853-6808418e-162d-4875-9add-1958d0b7c9dd.png)

And learn the code helper:

![image](https://user-images.githubusercontent.com/44065187/201734277-eb6183c7-5a76-4198-9d74-e0d54fc46839.png)

**Reply by pacmano1** [reply]
Assembling the pieces:

`$c('formatted_date',DateUtil.convertDate('yyyyMMdd', 'MM-dd-yyyy', msg['PID']['PID.7']['PID.7.1'].toString()))`

You don't have to drag drop of course to the mapper, you can just drag drop to a new empty JS step to get the value needed for the string.

**Reply by siddharthsharma9258** [reply]
> "19885425" to "1988-54-25"

That's not even a valid date.

**Reply by 5011-5011** [reply]
@pacmano1 **thaks so much for the orientation**

**Reply by 5011-5011** [reply]
@siddharthsharma9258 

> > "19885425" to "1988-54-25"
> 
> That's not even a valid date.

**Of course it's not real, it's an example.**

**Reply by pacmano1** [reply]
@siddharthsharma9258 should have posted a cool way to check if the date was valid!

BTW @5011-5011 many SQL engines are very smart about the date format, below is postgres.  That means you don't need to make mirth futz the date to do the insert.

![image](https://user-images.githubusercontent.com/44065187/201764342-45cd47ea-56b4-48ec-8d33-c21de2fc63ba.png)

![image](https://user-images.githubusercontent.com/44065187/201764431-1700f96f-62c2-47de-87c4-7ee8728b17fa.png)

![image](https://user-images.githubusercontent.com/44065187/201764536-32c3338b-ba30-4c34-a464-f82efb56e3e1.png)

**Reply by siddharthsharma9258** [reply]
Shouldn't we be using timestamp to store dates?

**Reply by pacmano1** [reply]
In many cases yes, but that is the implementer's decision.

**Reply by siddharthsharma9258** [reply]
Interesting. :)
![image](https://user-images.githubusercontent.com/20477452/201795129-842441f9-e76c-44c5-a951-43d5ad931898.png)
